ogr/factory.py

```diff
@@ -117,4 +117,46 @@ def get_service_class(
     for service, service_kls in mapping.items():
         if service in url:
             return service_kls
+
+    return None
+
+
+def get_service_class(
+    url: str, service_mapping_update: Dict[str, Type[GitService]] = None
+) -> Type[GitService]:
+    """
+    Get the matching service class from the url.
+
+    :param url: str (url of the project, e.g. "https://github.com/packit-service/ogr")
+    :param service_mapping_update: custom mapping from  service url (str) to service class
+    :return: Matched class (subclass of GitService)
+    """
+    service_kls = get_service_class_or_none(
+        url=url, service_mapping_update=service_mapping_update
+    )
+    if service_kls:
+        return service_kls
     raise OgrException("No matching service was found.")
+
+
+def get_instances_from_dict(instances: dict) -> Set[GitService]:
+    services = []
+    for key, value in instances.items():
+        service_kls = get_service_class_or_none(url=key)
+        if not service_kls:
+            if "type" not in value:
+                raise OgrException(f"No matching service was found for url '{key}'.")
+            service_type = value["type"]
+            if service_type not in _SERVICE_MAPPING:
+                raise OgrException(
+                    f"No matching service was found for type '{service_type}'."
+                )
+
+            service_kls = _SERVICE_MAPPING[service_type]
+            value.setdefault("instance_url", key)
```

Comment:
but `key` is not an URL, it's a hostname - I know that the proposed approach will work, but let's be a bit more consistent

Reply:
Not sure what you want:
- rename of the `instance_url` => problems with deprecation
- documentation, that the key has to be hostname

Reply:
honestly, I was literally thinking of doing "https://github.com/" instead of "github.com"

feel free to ignore if you think it's a bad idea or doesn't make sense

it does to me since it's explicit, "github.com" may even imply it's not an url

Reply:
okay, let's leave it as it is now